bazel_tools/scala.bzl

```diff
@@ -652,3 +654,77 @@ def da_scala_benchmark_jmh(
     deps = resolve_scala_deps(deps, scala_deps, versioned_scala_deps)
     runtime_deps = resolve_scala_deps(runtime_deps, scala_runtime_deps)
     _wrap_rule_no_plugins(scala_benchmark_jmh, deps, runtime_deps, **kwargs)
+
+def to_camel_case(name):
+    return "".join([part.capitalize() for part in name.split("_")])
+
+def da_scala_dar_resources_library(
+        daml_root_dir,
+        daml_dir_names,
+        lf_versions,
+        add_maven_tag = False,
+        maven_name_prefix = "",
+        **kwargs):
+    """
+    Define a Scala library with dar files as resources.
+    """
+    for lf_version in lf_versions:
+        for daml_dir_name in daml_dir_names:
+            # 1. Compile daml files
+            daml_compile_name = "%s-tests-%s" % (daml_dir_name, lf_version)
+            daml_compile_kwargs = {
+                "project_name": "%s-tests" % daml_dir_name,
+                "srcs": native.glob(["%s/%s/*.daml" % (daml_root_dir, daml_dir_name)]),
+                "target": lf_version,
+            }
+            daml_compile_kwargs.update(kwargs)
+            daml_compile(name = daml_compile_name, **daml_compile_kwargs)
+
+        # 2. Generate lookup objects
+        genrule_name = "test-dar-lookup-%s" % lf_version
+        genrule_command = """
+cat > $@ <<EOF
+package com.daml.ledger.test
+
+sealed trait TestDar { val path: String }
+
+object TestDar {
+  val paths: List[String] = List(
+EOF
+""" + "\n".join(["""
+echo "    \\"%s/%s-tests-%s.dar\\"," >> $@
+""" % (native.package_name(), test_name, lf_version) for test_name in daml_dir_names]) + """
+echo "  )\n}\n" >> $@
+""" + "\n".join(["""
+echo "case object %sTestDar extends TestDar { val path = \\"%s/%s-tests-%s.dar\\" }" >> $@
```

Comment:
This magic is why we pay you the big bucks. :moneybag:

Reply:
If you see magic anywhere it might be a bad sign that something needs to be changed 😃